cli/src/commands/branch.rs

```diff
@@ -100,9 +100,13 @@ pub struct BranchDeleteArgs {
 pub struct BranchListArgs {
     /// Show all tracking and non-tracking remote branches including the ones
     /// whose targets are synchronized with the local branches.
-    #[arg(long, short, conflicts_with_all = ["names", "revisions"])]
+    #[arg(long, short, conflicts_with_all = ["names", "revisions", "tracked"])]
     all: bool,
 
+    /// Show remote tracked branches for specified remotes
+    #[arg(long, short, conflicts_with_all = ["names", "revisions", "all"], value_name = "REMOTE")]
```

Comment:
I don't think `--tracked` and the other name filters should conflict. `--tracked=origin foo` is reasonable use case.

fwiw, that also applies to `--all` technically, but `--all [NAME]..` is disabled just because filtered result doesn't sound like "all". iirc, @ilyagr have some idea to unblock it.